lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
> But you'd still have the issue of ensuring the transaction is properly torn down in case of error

Actually, your `finalQuery` facility would probably be enough to ensure that happens.